cmake/modules/version.cmake

```diff
@@ -63,7 +63,7 @@ foreach(type file IN ZIP_LISTS VERSION_TYPE VERSION_FILE)
   string(REGEX MATCH "VERSION_TWEAK = ([0-9]*)" _ ${ver})
   set(${type}_VERSION_TWEAK ${CMAKE_MATCH_1})
 
-  string(REGEX MATCH "EXTRAVERSION = ([a-z0-9]*)" _ ${ver})
+  string(REGEX MATCH "EXTRAVERSION = ([a-z0-9\.]*)" _ ${ver})
```

Comment:
should we consider to allow a hyphen `-` as well.
That's is valid in pre-release version.
https://semver.org/#spec-item-9
> A pre-release version MAY be denoted by appending a hyphen and a series of dot separated identifiers immediately following the patch version. Identifiers MUST comprise only ASCII alphanumerics and hyphens [0-9A-Za-z-].
> ...
> Examples: 1.0.0-alpha, 1.0.0-alpha.1, 1.0.0-0.3.7, 1.0.0-x.7.z.92, **1.0.0-x-y-z.--**.

Check also: https://regex101.com/r/vkijKf/1/

Reply:
I wasn't aware of that. I can add it to allow that format?

Reply:
If you feel like updating PR with it

Reply:
I think it would be nice, else we probably just end back later with someone else missing the hyphen `-` in the format.

Reply:
I've added it.